recipes/imagemagick/all/conanfile.py

```diff
@@ -0,0 +1,399 @@
+        extracted_dir = "VisualMagick-" + visualmagick_version
+        os.rename(extracted_dir, "VisualMagick")
+
+    def build(self):
+        if self._is_msvc:
+            self._build_msvc()
+        else:
+            with tools.chdir(self._source_subfolder):
+                env_build = self._build_configure()
+                env_build.make()
+
+    def _build_msvc(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, {}):
+            tools.patch(**patch)
+
+        # FIXME: package LiquidRescale  aka liblqr
+        tools.replace_in_file(os.path.join('VisualMagick', 'lqr', 'Config.txt'),
+                              '#define MAGICKCORE_LQR_DELEGATE', '')
+        # FIXME: package LibRaw
+        tools.replace_in_file(os.path.join('VisualMagick', 'libraw', 'Config.txt'),
+                              '#define MAGICKCORE_RAW_R_DELEGATE', '')
+
+        # FIXME: package FLIF (FLIF: Free Lossless Image Format)
+        tools.replace_in_file(os.path.join('VisualMagick', 'flif', 'Config.txt'),
+                              '#define MAGICKCORE_FLIF_DELEGATE', '')
+
+        # FIXME: package libheif (High Efficiency Image File Format)
+        tools.replace_in_file(os.path.join('VisualMagick', 'libheif', 'Config.txt'),
+                              '#define MAGICKCORE_HEIC_DELEGATE', '')
+
+        # FIXME: package librsvg
+        tools.replace_in_file(os.path.join('VisualMagick', 'librsvg', 'Config.txt'),
+                              '#define MAGICKCORE_RSVG_DELEGATE', '')
+
+        if not self.options.shared:
+            for module in self._modules:
+                tools.replace_in_file(os.path.join('VisualMagick', module, 'Config.txt'), '[DLL]', '[STATIC]')
+            tools.replace_in_file(os.path.join('VisualMagick', 'coders', 'Config.txt'), '[DLLMODULE]',
+                                  '[STATIC]\n[DEFINES]\n_MAGICKLIB_')
+
+        if self.settings.arch == 'x86_64':
+            project = os.path.join('VisualMagick', 'configure', 'configure.vcxproj')
+            tools.replace_in_file(project, 'Win32', 'x64')
+            tools.replace_in_file(project, '/MACHINE:I386', '/MACHINE:x64')
+
+        with tools.chdir(os.path.join('VisualMagick', 'configure')):
+
+            toolset = self.settings.get_safe("compiler.toolset")
+            if not toolset:
+                toolset = {'12': 'v120',
+                           '14': 'v140',
+                           '15': 'v141'}.get(str(self.settings.compiler.version))
+            tools.replace_in_file('configure.vcxproj',
+                                  '<PlatformToolset>v120</PlatformToolset>',
+                                  '<PlatformToolset>%s</PlatformToolset>' % toolset)
+
+            msbuild = MSBuild(self)
+            # fatal error C1189: #error:  Please use the /MD switch for _AFXDLL builds
+            msbuild.build_env.flags = ["/MD"]
```

Comment:
Doesn't this conflict when `self.settings.compiler.runtime` is `MT`?

Reply:
I'm not sure I've never worked with windows builds and this currently doesn't work with them due to MFC.

Reply:
We'll handle this when/if the c3i build system gets MFC installed.

Reply:
yes, we can come back later

Reply:
> I've never worked with windows builds

Is the code provided function on your system? It seems odd to merge a lot of complicated logic if it's not somewhat working

Reply:
the recipe is from bincrafters.
the code to support Windows was working and it is still working locally.
however, CI doesn't provide ATL/MFC yet, which doesn't allow to build some projects on Windows (include ImageMagick).
we're working towards it in CI side right now, but it will take quite a time (we're changing the way we provision Windows agents, so some preparation work is needed, which is invisible from the outside).

for the recipe, it's not author's fault, but a CI fault that it doesn't provide MFC right away. let's keep this logic, as we will need to re-add it anyway.

Reply:
Perfect! I agree